### PR TITLE
fix(core): improve memory allocation issues in calling into icu 🙀

### PR DESCRIPTION
**Pull request by srl295**
For: #9121

- improves assert calls

@keymanapp-test-bot skip

### CODE REVIEW

**Review by srl295 — COMMENTED** [review]
Should I do anything besides DebugLog() in these cases?  The match will fail, in the case of a bad regex or a malloc failure.

File: core/src/ldml/ldml_transforms.cpp [hunk 1]
```diff
@@ -368,11 +368,12 @@ reorder_group::apply(std::u32string &str) const {
   // TODO-LDML: for now, assume matches entire string.
   // A needed optimization here would be to detect a common substring
   // at the end of the old and new strings, and keep the match_len
-  // minimal. This reduces thrash in core's context.
+  // minimal. This coudl reduce thrash in core's context.
```
Comment (jahorton) [inline core/src/ldml/ldml_transforms.cpp:371]:
```suggestion
  // minimal. This could reduce thrash in core's context.
```

**Review by mcdurdin — APPROVED** [review]

**Comment by mcdurdin** [comment]
> Should I do anything besides DebugLog() in these cases? The match will fail, in the case of a bad regex or a malloc failure.

No, just DebugLog is fine, because this not something that is user-resolveable.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.229-alpha](https://keyman.com/downloads/releases/alpha/17.0.229)

**Pull request merged**